src/Pact/Types/Term.hs

```diff
@@ -1214,11 +1207,31 @@ pattern TLitInteger i <- TLiteral (LInteger i) _
 pattern TLitBool :: Bool -> Term t
 pattern TLitBool b <- TLiteral (LBool b) _
 
-
 tLit :: Literal -> Term n
 tLit = (`TLiteral` def)
 {-# INLINE tLit #-}
 
+$( return [] )
```

Comment:
Agree with emily, this bit of gobbledygook needs a code comment.

Reply:
Agreed that it's very unfortunate that this is required. And it took me a while digging through GHC issue threads until I found this.

I'll add a comment.

Reply:
The more I look at this the more confused I get. Is there some bug in GHC 8.10 that requires this? If it's really necessary that might be a reason to not do this now, as whatever nonsense requires this might get fixed later.

Reply:
No, there are "bugs" in previous GHC versions. My understanding is that they had to change how TH compilations are staged as a consequence of doing cleanup/fixes in the type checker in GHC 8.10. It seems that old versions where too lenient.

Reply:
If I remember correctly, it's because of the way how GHC resolves constraints. Before it was doing it more lazy, which caused it accept impredicative constraints. Modern versions are more strict in forbidding impredicativity.